### PR TITLE
Use URL objects for aiohttp requests

### PR DESCRIPTION
**Pull request by bdraco**
Passing URL objects avoids having to have aiohttp create them each request Its a tiny speed up to getting camera snapshots

### CODE REVIEW

File: pyunifiprotect/api.py [hunk 1]
```diff
@@ -176,11 +177,12 @@ def __init__(
         if session is not None:
             self._session = session
 
-    @property
-    def base_url(self) -> str:
```
Comment (AngellusMortis) [inline pyunifiprotect/api.py:180]: Removing this is a breaking change.

Comment (AngellusMortis) [reply pyunifiprotect/api.py:180]: Nevermind, I see you are setting it statically below. Ignore me.

File: pyunifiprotect/api.py [hunk 2]
```diff
@@ -248,16 +250,18 @@ async def request(
         if require_auth:
             await self.ensure_authenticated()
 
-        url = urljoin(self.base_url, url)
+        request_url = self._url.joinpath(url[1:])
```
Comment (bdraco) [inline pyunifiprotect/api.py:253]: This wants the relative path so we have to trim the leading `/`

File: tests/test_api.py [hunk 3]
```diff
@@ -161,7 +161,7 @@ def test_base_url(protect_client: ProtectApiClient):
 
     protect_client._port = 443
 
-    assert protect_client.base_url == "https://127.0.0.1"
```
Comment (AngellusMortis) [inline tests/test_api.py:164]: We should either add a handler for `_port`/`_host` so if they are updated somehow, `base_url` is updated, or we should remove them completely (and by extension probably this test).

Comment (bdraco) [reply tests/test_api.py:164]: I don't see much of a use case for changing them later so I'll remove them and the test

Comment (bdraco) [reply tests/test_api.py:164]: index = self.bootstrap.nvr.hosts.index(self._host)

Looks like we can't remove it

File: tests/test_api.py [hunk 4]
```diff
@@ -160,6 +160,7 @@ def test_base_url(protect_client: ProtectApiClient):
     assert protect_client.ws_url == f"wss://127.0.0.1:0{arg}"
 
     protect_client._port = 443
+    protect_client._update_url()
```
Comment (bdraco) [inline tests/test_api.py:163]: Since we are reaching inside the object to change to 443 I figure this is ok as well

File: tests/test_api.py (hunk 3 above)
Comment (AngellusMortis) [reply tests/test_api.py:164]: You could store the result of that call every time the bootstrap if refreshed. Convert the yarl obj back to just the str host, get the index and store it locally.

Comment (bdraco) [reply tests/test_api.py:164]: Its only the snapshot urls that get called with any frequency so I left it as-is

Comment (AngellusMortis) [reply tests/test_api.py:164]: thumbnails are also called quite a lot if you load the media browser

**Review by AngellusMortis — APPROVED** [review]

**Pull request merged**